TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmode/TeleOp.java

```diff
@@ -3,27 +3,26 @@
 import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
 import com.qualcomm.robotcore.eventloop.opmode.OpMode;
 
-import org.firstinspires.ftc.teamcode.commands.drivetrain.ArcadeDrive;
 import org.firstinspires.ftc.teamcode.hardware.RobotHardware;
 import org.firstinspires.ftc.teamcode.subsystems.Drivetrain;
 
 @com.qualcomm.robotcore.eventloop.opmode.TeleOp (name = "Teste")
-public class TeleOp extends LinearOpMode {
-
-    private Drivetrain drivetrain;
+public class TeleOp extends OpMode {
 
     @Override
-    public void runOpMode() throws InterruptedException {
+    public void init() {
         RobotHardware.setHardwareMap(hardwareMap);
-        RobotHardware.initAll();
-        this.drivetrain = new Drivetrain();
-        while (opModeIsActive()) {
-            loopRobot();
-        }
+        RobotHardware.init();
     }
 
-    public void loopRobot() {
-        new ArcadeDrive(drivetrain, -gamepad1.left_stick_y, gamepad1.right_stick_x);
+    @Override
+    public void loop() {
+        Drivetrain.getInstance().arcadeDrive(-gamepad1.left_stick_y, gamepad1.right_stick_x);
```

Comment:
one thing that I would like to see is to define properly this gamepad1. Can you see if this global variable exists just in a OpMode? If not, i would be cool to create a class for controls, you also can see how TechZeus FTC works with this.

My suggestion is create a class with static functions that returns the values of gamepad, like DriverGamepad.getLeftStickY(), because I think we dont have IDE auto complete with this gamepad1,

Reply:
The gamepads are global variables just in opmodes. But we can implement a class for the controls, we just need to define a method on RobotHardware that instances the gamepad, that way in controls class we just call for that instance.